src/WebJobs.Script.WebHost/Handlers/ExceptionProcessingHandler.cs

```diff
@@ -0,0 +1,143 @@
+            {
+                { typeof(CryptographicException), CryptographicExceptionHandler }
+            };
+
+            return handlers;
+        }
+
+        public override void Handle(ExceptionHandlerContext context)
+        {
+            var error = new ApiErrorModel(HttpStatusCode.InternalServerError);
+
+            AuthorizationLevel currentLevel = context.Request.GetAuthorizationLevel();
+            foreach (var handler in GetExceptionHandlers(context.Exception))
+            {
+                handler(context.ExceptionContext, currentLevel, error);
+            }
+
+            TraceErrorEvent(context.ExceptionContext, error);
+
+            context.Result = new ResponseMessageResult(context.Request.CreateResponse(error.StatusCode, error));
+        }
+
+        private IEnumerable<ExceptionProcessor> GetExceptionHandlers(Exception exc)
+        {
+            if (exc == null)
+            {
+                yield break;
+            }
+
+            // We return our default handler first.
+            // Subsequent handlers can override everything done by it.
+            yield return DefaultExceptionHandler;
+
+            Type exceptionType = exc.GetType();
+            ExceptionProcessor exceptionHandler = null;
+            if (exceptionType != null && _handlers.TryGetValue(exceptionType, out exceptionHandler))
+            {
+                yield return exceptionHandler;
+            }
+        }
+
+        private static void DefaultExceptionHandler(ExceptionContext exceptionContext, AuthorizationLevel currentLevel, ApiErrorModel error)
+        {
+            if (currentLevel == AuthorizationLevel.Admin || exceptionContext.RequestContext.IsLocal)
+            {
+                error.Message = GetExceptionMessage(exceptionContext.Exception);
+
+                if (exceptionContext.RequestContext.IncludeErrorDetail)
+                {
+                    error.ErrorDetails = ExceptionFormatter.GetFormattedException(exceptionContext.Exception);
+                }
+            }
+            else
+            {
+                error.Message = $"An error has occurred. For more information, please check the logs for error ID {error.Id}";
+            }
+        }
+
+        private void TraceErrorEvent(ExceptionContext exceptionContext,  ApiErrorModel error)
+        {
+            string controllerName = exceptionContext.ControllerContext?.ControllerDescriptor.ControllerName ?? "<unknown>";
+            string actionName = exceptionContext.ActionContext?.ActionDescriptor.ActionName ?? "<unknown>";
+
+            string message = JsonConvert.SerializeObject(error);
+            var traceEvent = new TraceEvent(TraceLevel.Error, message, $"ApiError.{controllerName}.{actionName}", exceptionContext.Exception);
+            _traceWriterLoader.Value.Trace(traceEvent);
+        }
+
+        private static void CryptographicExceptionHandler(ExceptionContext exceptionContext, AuthorizationLevel currentLevel, ApiErrorModel error)
```

Comment:
If the user's function is doing crypto and this is thrown, we won't be emitting this keys specific log, right? I don't think so, because once we're running user code, any errors will be wrapped in a function exception type and handled?